packages/store/addon/-private/system/model/shim-model-class.ts

```diff
@@ -57,25 +57,31 @@ export default class ShimModelClass implements ModelSchema {
     return mapFromHash(relationships);
   }
 
-  eachAttribute<T>(callback: (this: T, key: string, attribute: AttributeSchema) => void, binding?: T) {
+  eachAttribute<T>(callback: (this: T | undefined, key: string, attribute: AttributeSchema) => void, binding?: T) {
     let attrDefs = this.__store._attributesDefinitionFor(this.modelName);
     Object.keys(attrDefs).forEach((key) => {
-      callback.call(binding, key, attrDefs[key]);
+      callback.call(binding, key, attrDefs[key] as AttributeSchema);
```

Comment:
This is just the usual issue with Object.keys, typescript has intentionally *not* typed it to narrow for keys just in case there are other props on the object which keys would see, in our case we are sure that this is not the case and so since the key exists the type is AttributeSchema and not undefined.